Quarkus example without any security extension fails to build even if `quarkus.hawtio.authenticationEnabled` to `false`

If the [quarkus example](https://github.com/hawtio/hawtio/tree/4.x/examples/quarkus) is set up as:

**pom.xml**
```diff
-    <!-- When authentication is enabled, a Quarkus security extension must be present -->
-    <dependency>
-      <groupId>io.quarkus</groupId>
-      <artifactId>quarkus-elytron-security-properties-file</artifactId>
-    </dependency>
```

**application.properties**
```diff
-#quarkus.hawtio.authenticationEnabled = false
+quarkus.hawtio.authenticationEnabled = false
... 
-quarkus.security.users.embedded.enabled = true
-quarkus.security.users.embedded.plain-text = true
-quarkus.security.users.embedded.users.hawtio = hawtio
-quarkus.security.users.embedded.roles.hawtio = admin
```

it fails to build with `mvn install` as follows:

```console
$ mvn install -pl examples/quarkus
...
[INFO] --- quarkus:3.8.2:build (default) @ hawtio-example-quarkus ---
[INFO] [io.quarkus.kubernetes.deployment.PropertyUtil] Kubernetes manifests are generated with 'The container port http' having default value '8080'. The app and manifests will get out of sync if the property 'quarkus.http.port' is changed at runtime.
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  12.862 s
[INFO] Finished at: 2024-03-13T15:12:22+09:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal io.quarkus.platform:quarkus-maven-plugin:3.8.2:build (default) on project hawtio-example-quarkus: Failed to build quarkus application: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
[ERROR] 	[error]: Build step io.quarkus.arc.deployment.ArcProcessor#validate threw an exception: jakarta.enterprise.inject.spi.DeploymentException: jakarta.enterprise.inject.UnsatisfiedResolutionException: Unsatisfied dependency for type io.quarkus.security.identity.IdentityProviderManager and qualifiers [@Default]
[ERROR] 	- injection target: io.hawt.quarkus.auth.HawtioQuarkusAuthenticator#identityProviderManager
[ERROR] 	- declared on CLASS bean [types=[io.hawt.quarkus.auth.HawtioQuarkusAuthenticator, java.lang.Object], qualifiers=[@Default, @Any], target=io.hawt.quarkus.auth.HawtioQuarkusAuthenticator]
[ERROR] 	at io.quarkus.arc.processor.BeanDeployment.processErrors(BeanDeployment.java:1508)
[ERROR] 	at io.quarkus.arc.processor.BeanDeployment.init(BeanDeployment.java:320)
[ERROR] 	at io.quarkus.arc.processor.BeanProcessor.initialize(BeanProcessor.java:160)
[ERROR] 	at io.quarkus.arc.deployment.ArcProcessor.validate(ArcProcessor.java:488)
[ERROR] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[ERROR] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
[ERROR] 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[ERROR] 	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
[ERROR] 	at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:849)
[ERROR] 	at io.quarkus.builder.BuildContext.run(BuildContext.java:256)
[ERROR] 	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1538)
[ERROR] 	at java.base/java.lang.Thread.run(Thread.java:840)
[ERROR] 	at org.jboss.threads.JBossThread.run(JBossThread.java:501)
[ERROR] Caused by: jakarta.enterprise.inject.UnsatisfiedResolutionException: Unsatisfied dependency for type io.quarkus.security.identity.IdentityProviderManager and qualifiers [@Default]
[ERROR] 	- injection target: io.hawt.quarkus.auth.HawtioQuarkusAuthenticator#identityProviderManager
[ERROR] 	- declared on CLASS bean [types=[io.hawt.quarkus.auth.HawtioQuarkusAuthenticator, java.lang.Object], qualifiers=[@Default, @Any], target=io.hawt.quarkus.auth.HawtioQuarkusAuthenticator]
[ERROR] 	at io.quarkus.arc.processor.Beans.resolveInjectionPoint(Beans.java:518)
[ERROR] 	at io.quarkus.arc.processor.BeanInfo.init(BeanInfo.java:638)
[ERROR] 	at io.quarkus.arc.processor.BeanDeployment.init(BeanDeployment.java:308)
[ERROR] 	... 13 more
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
```

--- a/platforms/quarkus/deployment/pom.xml
+++ b/platforms/quarkus/deployment/pom.xml
@@ -79,6 +79,7 @@
           <systemPropertyVariables>
             <java.util.logging.manager>org.jboss.logmanager.LogManager</java.util.logging.manager>
             <quarkus.log.level>DEBUG</quarkus.log.level>
+            <!--suppress UnresolvedMavenProperty -->
             <maven.home>${maven.home}</maven.home>
           </systemPropertyVariables>
         </configuration>
@@ -94,6 +95,7 @@
             <configuration>
               <systemPropertyVariables>
                 <java.util.logging.manager>org.jboss.logmanager.LogManager</java.util.logging.manager>
+                <!--suppress UnresolvedMavenProperty -->
                 <maven.home>${maven.home}</maven.home>
               </systemPropertyVariables>
             </configuration>

--- a/platforms/quarkus/runtime/src/main/java/io/hawt/quarkus/auth/HawtioQuarkusAuthenticator.java
+++ b/platforms/quarkus/runtime/src/main/java/io/hawt/quarkus/auth/HawtioQuarkusAuthenticator.java
@@ -4,17 +4,21 @@ import java.util.Arrays;
 import java.util.Optional;
 
 import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.enterprise.inject.Produces;
 import jakarta.inject.Inject;
 
 import io.hawt.system.AuthenticateResult;
 import io.hawt.util.Strings;
 import io.hawt.web.auth.AuthenticationConfiguration;
 import io.hawt.web.auth.AuthenticationThrottler;
+import io.quarkus.arc.DefaultBean;
 import io.quarkus.security.AuthenticationFailedException;
 import io.quarkus.security.credential.PasswordCredential;
 import io.quarkus.security.identity.IdentityProviderManager;
 import io.quarkus.security.identity.SecurityIdentity;
+import io.quarkus.security.identity.request.AuthenticationRequest;
 import io.quarkus.security.identity.request.UsernamePasswordAuthenticationRequest;
+import io.smallrye.mutiny.Uni;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
@@ -75,5 +79,31 @@ public class HawtioQuarkusAuthenticator {
 
         String[] roles = roleConfig.split(",");
         return Arrays.stream(roles).anyMatch(identity.getRoles()::contains);
+    }
+
+    /**
+     * Default no-op identity provider manager.
+     * <p>
+     * It is used only when Hawtio authentication is disabled, but should never
+     * be invoked from anywhere at runtime.
+     * <p>
+     * When Hawtio authentication is enabled a security capability is required,
+     * so the user will need to provide a proper identity provider manager other
+     * than this no-op one to boot the application.
+     */
+    @Produces
+    @DefaultBean
+    public IdentityProviderManager noopIdentityProviderManager() {
+        return new IdentityProviderManager() {
+            @Override
+            public Uni<SecurityIdentity> authenticate(AuthenticationRequest request) {
+                return null;
+            }
+
+            @Override
+            public SecurityIdentity authenticateBlocking(AuthenticationRequest request) {
+                return null;
+            }
+        };
     }
 }

